### DISCUSSION TITLE
Hybrid tables - is it possible with TE3?

### DISCUSSION THREAD

**Discussion by Stewwe1** [TabularEditor/TabularEditor3#410] [🙋♀️TE3 Q&A] [answered]
Dear community,

I have a question:

Is it possible to add a directquery real time table to an existing partitioning with the tabular editor as mentioned in this article?
https://powerbi.microsoft.com/en-us/blog/announcing-public-preview-of-hybrid-tables-in-power-bi-premium/

Thank you for the help!

Stewwe

**Comment by otykier** [comment, marked as answer]
Sure - Tabular Editor gives you full control over which tables and partitions to use (with the one exception currently being the "Entity"-type partitions used by DQ over AS, which is still a preview feature). So if you're working on a Power BI dataset with a high enough compatibility level, you can add a DirectQuery partition to a table that otherwise uses Import partitions. As Kay writes in the blog:

> But wait, there is more! Hybrid tables support is also available without an incremental refresh policy! For example, you can partition a table programmatically via XMLA endpoints to apply any partitioning scheme you like. For example, you could add a DirectQuery partition to a table for old (and possibly very large) historic data.

One limitation currently in Tabular Editor, is that we cannot set the RefreshPolicyMode, as illustrated in the code sample in the blog post. This will be possible with the next update.

**Reply by Stewwe1** [reply]
Thank you very much. Then we´ll wait for the next update :-)
